src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -134,17 +135,17 @@ def get_attrs(
                     ).get("uuid")
             alert_rule_triggers.append(serialized)
 
-        # Query for all alert rules once more to get the latest 10 activations
-        # last_10_activations_subquery = AlertRuleActivations.objects.filter(
-        #     alert_rule_id=OuterRef("id")
-        # ).order_by("-date_added")[:10]
-
-        # # Filter alerts by id and annotate with the last 10 activations
-        # alert_rules_with_activations = AlertRule.objects.filter(id__in=alert_rules.keys()).annotate(
-        #     last_activations=list(last_10_activations_subquery)
-        # )
-        # for alert_rule in alert_rules_with_activations:
-        #     result[alert_rule]["activations"] = serialize(alert_rule.last_activations, **kwargs)
+        alert_activations_ranked = AlertRuleActivations.objects.annotate(
+            rank=Window(
```

Comment:
This solution seems decent!

we're now ranking each activation by recency
and then below we're filtering that rank to those below 10.

this still produces a single query!